modules/cache.js

```diff
@@ -0,0 +1,166 @@
+var async = require('async');
+var transactionTypes = require('../helpers/transactionTypes.js');
+var client, self, logger, cacheEnabled,
+	cacheReady = true, errorCacheDisabled = 'Cache Unavailable';
+
+// Constructor
+function Cache (cb, scope) {
+	self = this;
+	client = scope.cache.client;
+	logger = scope.logger;
+	cacheEnabled = scope.cache.cacheEnabled;
+	setImmediate(cb, null, self);
+}
+
+Cache.prototype.isConnected = function () {
+	return cacheEnabled && client.connected;
+};
+
+Cache.prototype.isReady = function () {
+	return cacheReady && self.isConnected();
+};
+
+Cache.prototype.getJsonForKey = function (key, cb) {
+	if (!self.isConnected()) { 
+		return cb(errorCacheDisabled); 
+	}
+	client.get(key, function (err, value) {
+		// parsing string to json
+		return cb(err, JSON.parse(value));
```

Comment:
If `err` is populated - `value` is `undefined`, yes? If so `JSON.parse(value)` will fail because `undefined` is not valid JSON.